### PR TITLE
hcp: fix hcp artifact extraction method

### PR DESCRIPTION
**Pull request by lbajolet-hashicorp**
With Packer 1.10.1 we started warning when a build failed to complete because of a potential incompatibility with the builder being used.

This led to cases in which the build failed for other reasons, and Packer would still warn of potential incompatibilities, even if the builder was in effect HCP compatible.

We attempted to fix this issue by introducing a new error type, and checks when we read the artifacts linked to a build, however this loop would fail when any one of the artifacts is not compatible with HCP Packer, leading to false failures.

To avoid this problem, we log incompatibilities to the verbose logger, and only signal the problem if all the artifacts could not be used to upload data to HCP Packer, in which case it's almost certain that if the build succeeded and no artifacts are registered to the build, that all the components used are not compatible with HCP, and should be reported as such to users.

### CODE REVIEW

**Review by sylviamoss — APPROVED** [review]
Nice catch! 🚀

**Comment by sylviamoss** [comment]
Is it possible to catch that in a unit test?

**Comment by lbajolet-hashicorp** [comment]
> Is it possible to catch that in a unit test?

That seems very probable, I'll take a look at this, good call!

File: internal/hcp/registry/types.bucket_test.go [hunk 1]
```diff
@@ -385,3 +390,123 @@ func TestReadFromHCLBuildBlock(t *testing.T) {
 		})
 	}
 }
+
+func TestCompleteBuild(t *testing.T) {
```
Comment (sylviamoss) [inline internal/hcp/registry/types.bucket_test.go:394]: Nice! 🚀

**Pull request merged**